src/main/java/com/synacy/gradprogram/spock/exercise/RefundService.java

```diff
@@ -4,14 +4,36 @@
 
 public class RefundService {
 
-  public BigDecimal calculateRefund() {
-    // TODO: Implement me. Full refund if cancel reason is due to damaged item.
-    //  Also full refund if the order was cancelled within 3 days of order date, else refund half of the total cost.
-    return null;
+  private final DateUtils dateUtils;
+  private final RefundRepository refundRepository;
+  private final OrderRepository orderRepository;
+
+  public RefundService(DateUtils dateUtils, RefundRepository refundRepository, OrderRepository orderRepository) {
+    this.dateUtils = dateUtils;
+    this.refundRepository = refundRepository;
+    this.orderRepository = orderRepository;
   }
 
-  private void createAndSaveRefundRequest() {
-    // TODO: Implement me. Creates a TO_PROCESS refund request and saves it to the database
+  private BigDecimal calculateRefund(Order order, CancelOrderRequest cancelOrderRequest) {
+    final double DIVISOR = 2.0;
```

Comment:
We practice putting constants as a class level variable :+1: You can also improve the variable name to make it more clear on what it is for.